presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -690,7 +694,13 @@ public HiveInsertTableHandle beginInsert(ConnectorSession session, ConnectorTabl
 
         checkTableIsWritable(table.get());
 
-        HiveStorageFormat hiveStorageFormat = extractHiveStorageFormat(table.get());
+        HiveStorageFormat hiveStorageFormat;
+        if (respectTableFormat) {
```

Comment:
Well, you can set the format on a table in a CTAS, and my interpretation of `respectTableFormat` was that when it's false we ignore the user and use whatever format is specified in the configs for writing.
In theory, you can do CTAS with format=ORC, partition_by=array['ds'], which would do a dynamically partitioned write and use up a ton of memory, so it seems like that's just as dangerous as an insert into an existing table